File: Backend/mapping/src/main/java/com/mhp/coding/challenges/mapping/models/dto/blocks/ArticleBlockDto.java
package com.mhp.coding.challenges.mapping.models.dto.blocks;

// implements Comparable so that Collections.sort() can be used
public class ArticleBlockDto implements Comparable<ArticleBlockDto> {
    public ArticleBlockDto() {}
   
    public ArticleBlockDto(ArticleBlockDto articleBlockDto) {
        this.setSortIndex(articleBlockDto.getSortIndex());
    }
    
    private int sortIndex;

    public int getSortIndex() {
        return sortIndex;
    }

    public void setSortIndex(int sortIndex) {
        this.sortIndex = sortIndex;
    }
    
    // this method is used to compare two ArticleBlockDtos for sorting
    @Override
	public int compareTo(ArticleBlockDto blockDto) {
	    if (this.getSortIndex() > blockDto.getSortIndex())
	        return 1;
	    else if (this.getSortIndex() < blockDto.getSortIndex())
	        return -1;
	    else
	        return 0;
    }
}

File: Backend/mapping-kotlin/src/main/kotlin/com/mhp/coding/challenges/mapping/models/dto/blocks/ArticleBlockDto.kt
package com.mhp.coding.challenges.mapping.models.dto.blocks

// implements Comparable so that Collections.sort() can be used
interface ArticleBlockDto : Comparable<ArticleBlockDto> {
    val sortIndex: Int
    
    // this method is used to compare two ArticleBlockDtos for sorting
    override fun compareTo(other: ArticleBlockDto): Int {
	    if (this.sortIndex > other.sortIndex)
	        return 1;
	    else if (this.sortIndex < other.sortIndex)
	        return -1;
	    else
	        return 0;
    }
}
